anchore_engine/services/policy_engine/__init__.py

```diff
@@ -376,15 +330,22 @@ def handle_grypedb_sync(*args, **kwargs):
     cycle_time = kwargs["mythread"]["cycle_timer"]
 
     while True:
-        try:
-            result = GrypeDBSyncTask().execute()
-
-            if result:
-                logger.info("Grype DB synced to local instance via handler")
-        #         TODO narrow scope of exceptions in handlers. see https://github.com/anchore/anchore-engine/issues/1005
-        except Exception:
-            logger.exception(
-                "Error encountered when running GrypeDBSyncTask from async monitor"
+        provider = get_provider(get_vulnerabilities_config())
+        if provider.lower() == "grype":
```

Comment:
Should this use the `SupportedProvider` enum from the config file to verify the provider as opposed to just a string check? Makes it easier to update in the future and more clear where these checks happen